### ISSUE TITLE
TypeChecker.isAssignableFromType does not consider nullability

### ISSUE THREAD

**Issue by leisim** [dart-lang/sdk#48426]
This tracker is for issues related to: Analyzer

I would expect `TypeChecker.isAssignableFromType(staticType)` to take the nullability of the types into account. It returns `true` incorrectly if the TypeChecker type is not nullable but `staticType` is nullable.

Dart SDK version: 2.16.1

**Comment by lrhn** [comment]
Is this the `TypeChecker` from `package:source_gen`? (If so, the issue would fit better in https://github.com/dart-lang/source_gen/issues).

**Comment by leisim** [comment]
@lrhn yes, sorry! Should I close here and reopen or can you move the issue?
